### PR TITLE
Updated decision trees

### PR DESCRIPTION
**Pull request by nielenventer**
Updates dtrees:
- No explicit return node (child idx indicates return)
- Clearer tree property names
- MLFeature properties (return float)

### CODE REVIEW

File: Tensile/Source/lib/include/Tensile/ContractionProblemProperties.hpp [hunk 1]
```diff
@@ -27,6 +27,7 @@
 #pragma once
 
 #include <Tensile/ContractionProblem.hpp>
+#include <Tensile/ContractionSolution.hpp>
```
Comment (benjaminulmer) [inline Tensile/Source/lib/include/Tensile/ContractionProblemProperties.hpp:30]: am I just missing why this include needed to be added?

**Review by benjaminulmer — COMMENTED** [review]
Overall looks good. My one piece of feedback would be it seems there's some overlap in functionality between the calculations you do in the new functions in ContractionSolution (numTiles0, numTiles1, and tileGranularity) and the existing computeGranularities. Did you look at all into leveraging the existing code here?

File: Tensile/Source/lib/include/Tensile/ContractionProblemProperties.hpp (hunk 1 above)
Comment (nielenventer) [reply Tensile/Source/lib/include/Tensile/ContractionProblemProperties.hpp:30]: good catch, that was a relic of when the feature properties were in this file

**Comment by nielenventer** [comment]
Made a mess by PRing my dev instead of just the trees branch, updated PR here:
https://github.com/ROCmSoftwarePlatform/Tensile/pull/1568

File: Tensile/Source/lib/include/Tensile/ContractionProblemProperties.hpp (hunk 1 above)
Comment (nielenventer) [reply Tensile/Source/lib/include/Tensile/ContractionProblemProperties.hpp:30]: R.e. overlap, I removed the tile calc functions (`numTiles<x>`) completely, they were too specific to MLFeature, and actually it seemed clearer to just have no function. `tileGranularity` was changed to `computeGranularity` and used in a few more places.

**Pull request closed without merging**